File: gnostica-server/src/main/java/com/gnostica/modules/auth/service/AuthService.java
package com.gnostica.modules.auth.service;

import com.gnostica.modules.auth.dto.request.LoginRequest;
import com.gnostica.modules.auth.dto.response.LoginResponse;
import com.gnostica.modules.auth.dto.request.RegisterRequest;
import com.gnostica.core.model.Account;

public interface AuthService {
    Account register(RegisterRequest request);
    LoginResponse login(LoginRequest request);
    LoginResponse loginWithGoogle(com.gnostica.modules.auth.dto.request.GoogleLoginRequest request);
    boolean verifyOTP(String email, String code);
    void resendVerificationEmail(String email);
    Account findByEmail(String email);
    void forgotPassword(String email);
    void resetPassword(String email, String code, String newPassword);
    void becomeInstructor(String email);
    
    // Quản lý Account cho Admin
    java.util.List<Account> getAllAccounts();
    org.springframework.data.domain.Page<Account> getAccountsByRole(String roleName, int page, int size);
    void lockAccount(java.util.UUID id, String reason);
    void unlockAccount(java.util.UUID id);
    void updateAvatar(String email, String avatarUrl);

    void updatePersonalization(String email, com.gnostica.modules.user.dto.request.PersonalizationDTO dto);
    void updateProfile(String email, com.gnostica.modules.auth.dto.request.ProfileUpdateRequest request);
    void changePassword(String email, String currentPassword, String newPassword);
}

File: gnostica-mobile/src/services/auth/authService.js
import api from '../../config/api';

const authService = {
    login: async (email, password) => {
        try {
            const response = await api.post('/auth/login', { email, password });
            return response;
        } catch (error) {
            throw error;
        }
    },

    googleLogin: async (payload) => {
        try {
            const response = await api.post('/auth/google', payload);
            return response;
        } catch (error) {
            throw error;
        }
    },

    getOAuth2User: async (email) => {
        try {
            const response = await api.get('/auth/user', { params: { email } });
            return response;
        } catch (error) {
            throw error;
        }
    },

    register: async (fullName, email, password) => {
        try {
            const response = await api.post('/auth/register', { 
                fullName, 
                email, 
                password 
            });
            return response;
        } catch (error) {
            throw error;
        }
    },

    verifyOTP: async (email, code) => {
        try {
            const response = await api.post(`/auth/verify`, null, {
                params: { email, code }
            });
            return response;
        } catch (error) {
            throw error;
        }
    },

    resendOTP: async (email) => {
        try {
            const response = await api.post(`/auth/resend-otp`, null, {
                params: { email }
            });
            return response;
        } catch (error) {
            throw error;
        }
    },

    forgotPassword: async (email) => {
        try {
            const response = await api.post(`/auth/forgot-password`, null, {
                params: { email }
            });
            return response;
        } catch (error) {
            throw error;
        }
    },

    resetPassword: async (email, code, newPassword) => {
        try {
            const response = await api.post('/auth/reset-password', { 
                email, 
                code, 
                newPassword 
            });
            return response;
        } catch (error) {
            throw error;
        }
    }
};

export default authService;

